evm/src/executor/backend/mod.rs

```diff
@@ -661,6 +703,62 @@ impl Backend {
         }
         Ok(())
     }
+
+    /// Returns the block numbers required for replaying a transaction
+    fn get_block_numbers_for_transaction(
+        &self,
+        id: LocalForkId,
+        transaction: H256,
+    ) -> eyre::Result<(U64, U64)> {
+        let fork = self.inner.get_fork_by_id(id)?;
+        let tx = fork.db.db.get_transaction(transaction)?;
+
+        // get the block number we need to fork
+        if let Some(tx_block) = tx.block_number {
+            // we need to subtract 1 here because we want the state before the transaction
+            // was mined
+            let fork_block = tx_block - 1;
+            Ok((fork_block, tx_block))
+        } else {
+            let block = fork
+                .db
+                .db
+                .get_full_block(BlockNumber::Latest)?
+                .number
```

Comment:
can you not do `eth_blockNumber` here instead of asking for the full block?

Reply:
can do that by adding another request variant for eth_blockNumber